src/topology/continuous_function/basic.lean

```diff
@@ -224,6 +224,13 @@ def restrict (f : C(α, β)) : C(s, β) := ⟨f ∘ coe⟩
 
 @[simp] lemma coe_restrict (f : C(α, β)) : ⇑(f.restrict s) = f ∘ coe := rfl
 
+@[simp] lemma restrict_apply (f : C(α, β)) (s : set α) (x : s) : f.restrict s x = f x := rfl
+
+/-- Do we want this one or not? --/
+@[simp] lemma restrict_apply_mk (f : C(α, β)) (s : set α) (x : α) (hx : x ∈ s) :
+  f.restrict s ⟨x, hx⟩ = f x :=
+rfl
+
```

Comment:
I'm ok with keeping this one if you have a mild preference for it; it's unusual, but it is actually useful for rewriting backwards. Make sure you remove the comment before merging
```suggestion
@[simp] lemma restrict_apply_mk (f : C(α, β)) (s : set α) (x : α) (hx : x ∈ s) :
  f.restrict s ⟨x, hx⟩ = f x :=
rfl

```

Reply:
OK, let's keep it. I've removed the comment.